test/reporter.spec.js

```diff
@@ -2,80 +2,64 @@
 /* eslint-env mocha */
 var expect = require('chai').expect
 var gutil = require('gulp-util')
-var mockery = require('mockery')
+var proxyquire = require('proxyquire')
 var sinon = require('sinon')
 
 describe('#reporter()', function () {
-  var gulpUtil
   var reporter
+  var mockGulpUtil
+  var mockReporter
   var stream
 
   describe('when the stream contains PugLinter errors', function () {
-    var streamFile = new gutil.File({
-      base: 'base',
-      contents: Buffer.from(''),
-      cwd: __dirname,
-      path: 'path.pug'
-    })
+    var streamFile
+
+    beforeEach(function () {
+      streamFile = new gutil.File({
+        base: 'base',
+        contents: Buffer.from(''),
+        cwd: __dirname,
+        path: 'path.pug'
+      })
 
-    streamFile.pugLinter = {errors: [{message: 'some error'}]}
+      streamFile.pugLinter = {errors: [{message: 'some error'}]}
+    })
 
     it('should print the error for default reporter', function () {
-      gulpUtil = {log: function () {}}
+      mockGulpUtil = {log: sinon.stub()}
 
-      sinon.stub(gulpUtil)
-
-      mockery.enable({
-        useCleanCache: true,
-        warnOnUnregistered: false
-      })
-
-      mockery.registerMock('gulp-util', gulpUtil)
-
-      reporter = require('../reporter')
+      reporter = proxyquire('../reporter', {'gulp-util': mockGulpUtil})
 
       stream = reporter()
 
       stream.write(streamFile)
 
       stream.end()
 
-      expect(gulpUtil.log.calledWith('some error'))
+      expect(mockGulpUtil.log.calledWith('some error'))
         .to.be.ok
-
-      gulpUtil.log.restore()
-
-      mockery.disable()
-
-      mockery.deregisterAll()
     })
 
     it('should report errors for named reporter', function () {
-      var funcReporter = function (errors) {}
-
-      var spiedReporter = sinon.spy(funcReporter)
+      funcReporter['@noCallThru'] = true
```

Comment:
See this comment for details: https://github.com/thlorenz/proxyquire/issues/207#issuecomment-398046679